nifi-nar-bundles/nifi-simulator-bundle/nifi-simulator-processors/src/main/java/com/apache/nifi/processors/simulator/GenerateTimeSeriesFlowFile.java

```diff
@@ -0,0 +1,180 @@
+            .name("Success")
+            .description("When the flowfile is successfully generated")
+            .build();
+
+    private List<PropertyDescriptor> descriptors;
+
+    private Set<Relationship> relationships;
+
+    @Override
+    protected void init(final ProcessorInitializationContext context) {
+
+        final List<PropertyDescriptor> descriptors = new ArrayList<>();
+
+        descriptors.add(SIMULATOR_CONFIG);
+        descriptors.add(PRINT_HEADER);
+
+        this.descriptors = Collections.unmodifiableList(descriptors);
+
+        final Set<Relationship> relationships = new HashSet<>();
+        relationships.add(SUCCESS);
+        this.relationships = Collections.unmodifiableSet(relationships);
+    }
+
+    @Override
+    public Set<Relationship> getRelationships() {
+        return this.relationships;
+    }
+
+    @Override
+    public final List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        return descriptors;
+    }
+
+    @Override
+    public void onPropertyModified(final PropertyDescriptor descriptor, final String oldValue, final String newValue) {
+        if (SIMULATOR_CONFIG.equals(descriptor))
+            simConfig = null;
+    }
+
+    @OnScheduled
+    public void onScheduled(final ProcessContext context) {
+        loadConfiguration(context.getProperty(SIMULATOR_CONFIG).getValue());
+    }
+
+    @Override
+    public void onTrigger(final ProcessContext context, final ProcessSession session) throws ProcessException {
+        ComponentLog logger = getLogger();
+        FlowFile flowFile = session.get();
+
+        // Create the flowfile, as it probably does not exist
+        if (flowFile == null)
+            flowFile = session.create();
+
+        // Get the data
+        String data = generateData(context.getProperty(PRINT_HEADER).asBoolean());
+
+        // Write the results back out to flow file
+        try{
+            flowFile = session.write(flowFile, out -> out.write(data.getBytes()));
+            session.getProvenanceReporter().create(flowFile);
+            session.transfer(flowFile, SUCCESS);
+        } catch (ProcessException ex) {
+            logger.error("Unable to write generated data out to flowfile. Error: ", ex);
+        }
+    }
+
+    // Loads the configuration from the file
+    private void loadConfiguration(String fileName){
+        if (simConfig == null){
+            // Load the simulator configuration
+            if (fileName.contains("/configs/unitTestConfig.json"))
+                isTest = true;
+            try{
+                simConfig = SimController.getConfiguration(fileName);
+            }catch (Exception ex){
+                getLogger().error("Error loading configuration: " + ex.getMessage());
+                throw ex;
+            }
+
+        }
+    }
+
+    // Actually do the data generation via TSimulus
+    private String generateData(boolean printHeader){
+        LocalDateTime queryTime = LocalDateTime.now();
+        if(isTest)
```

Comment:
Testing logic should not be hard-coded in like this.

Reply:
I would agree would agree with Mike here. In the one odd chance a user happens to have the same file name and location things would break for them.  It looks as if you are seeding a particular time which I believe is being compared later within your unit test.  One suggestion would be if there is a particular use case where users may want to provide a seed value for queryTime, expose that as a property.  That way your unit test could leverage it as well as users, and it would eliminate a need to have test specific logic.

Reply:
I completely agree, should have the commit later today.